File: web-app/tests/WatchListRepository.test.ts
/* eslint-env browser */
import { describe, it, expect, beforeEach } from 'vitest';
import { WatchListRepository } from '../src/repositories/WatchListRepository';

describe('WatchListRepository', () => {
  beforeEach(() => {
    localStorage.clear();
  });

  it('saves and loads symbols', async () => {
    const repo = new WatchListRepository();
    await repo.save(['AAPL', 'GOOG']);
    const list = await repo.list();
    expect(list).toEqual(['AAPL', 'GOOG']);
  });

  it('returns empty array when no data', async () => {
    const repo = new WatchListRepository();
    const list = await repo.list();
    expect(list).toEqual([]);
  });

  it('rejects invalid list', async () => {
    const repo = new WatchListRepository();
    // @ts-expect-error
    await expect(repo.save(['AAPL', ''])).rejects.toThrow();
  });
});

File: mobile-app/test/watch_list_repository_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:shared_preferences/shared_preferences.dart';
import 'package:mobile_app/repositories/watch_list_repository.dart';

void main() {
  TestWidgetsFlutterBinding.ensureInitialized();

  group('WatchListRepository', () {
    setUp(() {
      SharedPreferences.setMockInitialValues({});
    });

    test('saves and loads symbols', () async {
      final repo = WatchListRepository();
      await repo.save(['AAPL', 'GOOG']);
      final list = await repo.list();
      expect(list, ['AAPL', 'GOOG']);
    });

    test('returns empty list when no data', () async {
      final repo = WatchListRepository();
      final list = await repo.list();
      expect(list, isEmpty);
    });

    test('rejects invalid list', () async {
      final repo = WatchListRepository();
      expect(() => repo.save(['AAPL', '']), throwsArgumentError);
    });
  });
}
